kotsadm/pkg/updatechecker/updatechecker.go

```diff
@@ -305,3 +290,62 @@ func CheckForUpdates(appID string, deploy bool) (int64, error) {
 
 	return availableUpdates, nil
 }
+
+func GetReportingInfo(appID string) (*upstreamtypes.ReportingInfo, error) {
+	r := upstreamtypes.ReportingInfo{}
+
+	downstreams, err := store.GetStore().ListDownstreamsForApp(appID)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to list downstreams for app")
+	}
+	r.ClusterID = downstreams[0].ClusterID
+
+	deployedAppSequence, err := downstream.GetCurrentParentSequence(appID, downstreams[0].ClusterID)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get current downstream parent sequence")
+	}
+
+	// info about the deployed app sequence
+	if deployedAppSequence != -1 {
+		deployedArchiveDir, err := store.GetStore().GetAppVersionArchive(appID, deployedAppSequence)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to get app version archive")
+		}
+
+		deployedKotsKinds, err := kotsutil.LoadKotsKindsFromPath(deployedArchiveDir)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to load kotskinds from path")
+		}
+
+		r.DownstreamCursor = deployedKotsKinds.Installation.Spec.UpdateCursor
+		r.DownstreamChannelID = deployedKotsKinds.Installation.Spec.ChannelID
+		r.DownstreamChannelName = deployedKotsKinds.Installation.Spec.ChannelName
+	}
+
+	// get kubernetes cluster version
+	cfg, err := config.GetConfig()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get cluster config")
+	}
+	clientset, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create kubernetes clientset")
+	}
+	k8sVersion, err := clientset.ServerVersion()
```

Comment:
What RBAC permissions does this need, if any?

Reply:
i'm not sure. @divolgin any idea?

Reply:
no additional RBAC permissions required to get this information. confirmed on a minimal RBAC installation on an existing cluster